src/ui/control/scale_control.js

```diff
@@ -104,26 +112,35 @@ function updateScale(map, container, options) {
         const maxFeet = 3.2808 * maxMeters;
         if (maxFeet > 5280) {
             const maxMiles = maxFeet / 5280;
-            setScale(container, maxWidth, maxMiles, map._getUIString('ScaleControl.Miles'), map);
+            setScale(container, maxWidth, maxMiles, language, 'mile', map);
         } else {
-            setScale(container, maxWidth, maxFeet, map._getUIString('ScaleControl.Feet'), map);
+            setScale(container, maxWidth, maxFeet, language, 'foot', map);
         }
     } else if (options && options.unit === 'nautical') {
         const maxNauticals = maxMeters / 1852;
-        setScale(container, maxWidth, maxNauticals, map._getUIString('ScaleControl.NauticalMiles'), map);
+        setScale(container, maxWidth, maxNauticals, language, 'nautical-mile', map);
     } else if (maxMeters >= 1000) {
-        setScale(container, maxWidth, maxMeters / 1000, map._getUIString('ScaleControl.Kilometers'), map);
+        setScale(container, maxWidth, maxMeters / 1000, language, 'kilometer', map);
     } else {
-        setScale(container, maxWidth, maxMeters, map._getUIString('ScaleControl.Meters'), map);
+        setScale(container, maxWidth, maxMeters, language, 'meter', map);
     }
 }
 
-function setScale(container, maxWidth, maxDistance, unit, map) {
+function setScale(container, maxWidth, maxDistance, language, unit, map) {
     const distance = getRoundNum(maxDistance);
     const ratio = distance / maxDistance;
+
     map._requestDomTask(() => {
         container.style.width = `${maxWidth * ratio}px`;
-        container.innerHTML = `${distance}&nbsp;${unit}`;
+
+        // Intl.NumberFormat doesn't support nautical-mile as a unit
+        if (unit === 'nautical-mile') {
+            container.innerHTML = `${distance}&nbsp;${map._getUIString('ScaleControl.NauticalMiles')}`;
+            return;
+        }
```

Comment:
Should we drop nautical miles from the API instead? This exception now sticks out like a sore thumb...

Reply:
Yep, I agree! But that would be a breaking change

Reply:
I'd suggest leaving nautical miles as a hardcoded english string. I doubt many (any?) users are using this, but this would avoid the breaking change and the localization file bloat. I'd be fine with removing it completely too.

I don't think we need a new display option here. It could be useful, but we should avoid increasing the api surface if we don't have a clear reason to do so.

Reply:
> I'd suggest leaving nautical miles as a hardcoded english string. I doubt many (any?) users are using this, but this would avoid the breaking change and the localization file bloat. I'd be fine with removing it completely too.

That looks like a good option to me too. Ideally we would remove it fully, but keeping as hardcoded English-only is a good compromise for keeping compatibility while not increasing the API surface.

Reply:
Yep, I agree! I'll hardcode the nautical miles as an English string then. Thanks!